components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/api/impl/BusinessRulesApiServiceImpl.java

```diff
@@ -0,0 +1,422 @@
+import org.wso2.carbon.business.rules.core.bean.scratch.BusinessRuleFromScratch;
+import org.wso2.carbon.business.rules.core.bean.template.BusinessRuleFromTemplate;
+import org.wso2.carbon.business.rules.core.exceptions.BusinessRuleNotFoundException;
+import org.wso2.carbon.business.rules.core.exceptions.RuleTemplateScriptException;
+import org.wso2.carbon.business.rules.core.exceptions.TemplateManagerServiceException;
+import org.wso2.carbon.business.rules.core.services.TemplateManagerService;
+import org.wso2.carbon.business.rules.core.util.TemplateManagerConstants;
+import org.wso2.carbon.business.rules.core.util.TemplateManagerHelper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.Response;
+
+/**
+ * Implementation of business rules REST API
+ */
+@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaMSF4JServerCodegen",
+        date = "2017-10-13T06:19:32.032Z")
+public class BusinessRulesApiServiceImpl extends BusinessRulesApiService {
+    private static final Logger log = LoggerFactory.getLogger(BusinessRulesApiServiceImpl.class);
+
+    @Override
+    public Response createBusinessRule(String businessRule, Boolean shouldDeploy
+    ) throws NotFoundException {
+        TemplateManagerService templateManagerService = TemplateManagerInstance.getInstance();
+        // convert the string received from API, as a json object
+        Gson gson = new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create();
+        JsonObject businessRuleJson = gson.fromJson(businessRule, JsonObject.class);
+        int status;
+        List<Object> responseData = new ArrayList<Object>();
+        String businessRuleName = null;
+        try {
+            // Check the business rule type of the json object
+            if (businessRuleJson.get("type").toString().equals("\"" + TemplateManagerConstants
+                    .BUSINESS_RULE_TYPE_TEMPLATE + "\"")) {
+                // Convert to business rule from template and create
+                BusinessRuleFromTemplate businessRuleFromTemplate = TemplateManagerHelper
+                        .jsonToBusinessRuleFromTemplate(businessRule);
+
+                businessRuleName = businessRuleFromTemplate.getName();
+
+                status = templateManagerService.createBusinessRuleFromTemplate(businessRuleFromTemplate, shouldDeploy);
+            } else {
+                BusinessRuleFromScratch businessRuleFromScratch = TemplateManagerHelper.jsonToBusinessRuleFromScratch
+                        (businessRule);
+                businessRuleName = businessRuleFromScratch.getName();
+
+                status = templateManagerService.createBusinessRuleFromScratch(businessRuleFromScratch, shouldDeploy);
+            }
+        } catch (TemplateManagerServiceException e) {
+            log.error("Failed to create business rule '" + businessRuleName + "' due to " +
```

Comment:
Since you are using SLF4J, use parameterized messages instead of string concatenation. Refer [SLF4J FAQ](https://www.slf4j.org/faq.html#logging_performance).

Reply:
Also, don't append the exception message to the log message. The logger will print the exception message when printing the exception.